javacord-core/src/main/java/org/javacord/core/util/handler/channel/thread/ThreadUpdateHandler.java

```
@@ -52,22 +73,138 @@ private void handleThread(final JsonNode jsonChannel) {
+                    new ServerThreadChannelChangeArchiveTimestampEventImpl(thread,
+                            newArchiveTimestamp, oldArchiveTimestamp);
+
+
+            api.getEventDispatcher().dispatchServerThreadChannelChangeArchiveTimestampEvent(
+                        (DispatchQueueSelector) thread.getServer(), thread.getServer(), thread, event);
+        }
+
+        final Set<ThreadMember> oldMembers = thread.getMembers();
+        final Set<ThreadMember> newMembers = new HashSet<>();
+        if (jsonChannel.hasNonNull("member")) {
+            // If userId is not included, that means this came from a GUILD_CREATE event
+            // This means the userId is the bot's and the thread id is from this thread
+            // See https://github.com/Javacord/Javacord/issues/898
+            if (jsonChannel.get("member").hasNonNull("user_id")) {
+                newMembers.add(new ThreadMemberImpl(api, server, jsonChannel.get("member")));
+            } else {
+                newMembers.add(new ThreadMemberImpl(api, server, jsonChannel.get("member"),
+                        thread.getId(), api.getYourself().getId()));
             }
         }
 
+        if (!Objects.deepEquals(oldMembers, newMembers)) {
+            thread.setMembers(newMembers);
+
+            final ServerThreadChannelChangeMembersEvent event =
+                    new ServerThreadChannelChangeMembersEventImpl(thread, newMembers, oldMembers);
+
+
+            api.getEventDispatcher().dispatchServerThreadChannelChangeMembersEvent(
+                        (DispatchQueueSelector) thread.getServer(), thread.getServer(), thread, event);
+        }
+
+        final int oldNumberOfMessages = thread.getTotalNumberOfMessagesSent();
+        final int newNumberOfMessages = jsonChannel.get("total_message_count").asInt(0);
+        if (oldNumberOfMessages != newNumberOfMessages) {
+            thread.setTotalNumberOfMessagesSent(newNumberOfMessages);
+        }
```

`total_message_count` does not exist (change to `total_message_sent`) so this should throw an exception. Did you try to run this code? 
You also need to dispatch the listener here